drape_frontend/drape_engine.cpp

```diff
@@ -31,25 +31,31 @@ DrapeEngine::DrapeEngine(Params && params)
   location::EMyPositionMode mode = params.m_initialMyPositionMode.first;
   if (!params.m_initialMyPositionMode.second && !settings::Get(settings::kLocationStateMode, mode))
   {
-    mode = location::MODE_UNKNOWN_POSITION;
+    mode = location::PendingPosition;
   }
-  else if (mode == location::MODE_ROTATE_AND_FOLLOW)
+  else if (mode == location::FollowAndRotate)
   {
     // If the screen rect setting in follow and rotate mode is missing or invalid, it could cause invalid animations,
     // so the follow and rotate mode should be discarded.
     m2::AnyRectD rect;
     if (!(settings::Get("ScreenClipRect", rect) && df::GetWorldRect().IsRectInside(rect.GetGlobalRect())))
-      mode = location::MODE_FOLLOW;
+      mode = location::Follow;
   }
 
+  double timeInBackground = 0.0;
+  double lastEnterBackground = 0.0;
+  if (settings::Get("LastEnterBackground", lastEnterBackground))
+    timeInBackground = my::Timer::LocalTime() - lastEnterBackground;
```

Comment:
Это обязательно нужно протестировать, выставив время на 1 января 1970 года и на какую-нибудь будущую дату.

> On 14.4.2016, at 11.32, Roman Kuznetsov <notifications@github.com> wrote:
> 
> In drape_frontend/drape_engine.cpp:
> 
> >    }
> >  
> > +  double timeInBackground = 0.0;
> > +  double lastEnterBackground = 0.0;
> > +  if (settings::Get("LastEnterBackground", lastEnterBackground))
> > +    timeInBackground = my::Timer::LocalTime() - lastEnterBackground;
> Функция может работать не как хочется и в более простых случаях, когда приложение покрэшилось или было убито в фоне :) Мы считаем это редкими случаями. Поэтому это допустимое поведение и согласовано с Артемом и Игорем. К тому же ничего плохого для пользователя не случится.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>